src/CMakeLists.txt

```diff
@@ -10,14 +10,16 @@ if(BUILD_LIBRARY)
 	return()
 endif()
```

Comment:
Please provide a proper explanation of what why are doing this change. Maybe split the changes in several patches and we can agree on some obvious moves which we can merge right away and leave others open for discussion.

I think the initial design of the source tree was to resemble with the Linux kernel and we should try to keep it like that. 

For example:
*  src/init/init.c → src/init.c.

I am not sure this change makes code more readable? User accustomed with Linux kernel will always look for a init directory. More than that in feature this directory might get some new files.

Maybe @lgirdwood can chime in.

F

Reply:
Not everything done in the Linux kernel is the right thing to do.
Our source code is as close to the kernel as it can be without introducing too much of a mess.
Of course I can live with init.c in init/init.c, but it would be logical to have main entry point of the firmware clearly visible by the new developers (not only with Linux experience).